### PR TITLE
Fix substitutions, add integration tests

### LINKED ISSUE #13: doesn't seem to be replacing project-name and authors
```
Ashleys-MacBook-Pro-2:cargo-generate ag_dubs$ cargo run -- --git https://github.com/ashleygwilliams/wasm-pack-template.git
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `target/debug/cargo-generate --git 'https://github.com/ashleygwilliams/wasm-pack-template.git'`
The project's name is: wasm-fart
Nice, I'll call your project `wasm-fart`
Done!
Ashleys-MacBook-Pro-2:cargo-generate ag_dubs$ ls
CODE_OF_CONDUCT.md	Cargo.lock		LICENSE-APACHE		README.md		target
CONTRIBUTING.md		Cargo.toml		LICENSE-MIT		src			wasm-fart
Ashleys-MacBook-Pro-2:cargo-generate ag_dubs$ cd wasm-fart/
Ashleys-MacBook-Pro-2:wasm-fart ag_dubs$ ls
Cargo.toml	LICENSE_APACHE	LICENSE_MIT	README.md	src
Ashleys-MacBook-Pro-2:wasm-fart ag_dubs$ cat Cargo.toml
[package]
name = {{project-name}}
version = "0.1.0"
authors = {{authors}}

[dependencies]
wasm-bindgen = "0.2"
```

### PR DESCRIPTION
**Pull request by killercup**
While trying to make substitution work (i.e., find a missing `!` basically), I wrote a test setup and a very simple test.

Fixes #13
First step towards #9

### CODE REVIEW

**Comment by killercup** [comment]
Build failed on Windows because

> stderr=```error: Error cleaning up cloned template
> caused by: Access is denied. (os error 5)

@ashleygwilliams do you have access to a windows machine to debug this? Cause I don't right now :(

**Comment by ashleygwilliams** [comment]
@killercup yeah @steveklabnik can you help with this?

File: tests/integration/basics.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+use super::helpers::*;
+
+use assert_cmd::prelude::*;
+use predicates::prelude::*;
+use std::process::Command;
+
+#[test]
+fn simple_cargo_thing() {
+    let template = dir("template")
+        .file(
```
Comment (ashleygwilliams) [inline tests/integration/basics.rs]: long run i'd love these as fixtures, inline files get messy quick (is fine for now, but just noting)

File: tests/integration/main.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+mod helpers {
+    use std::env;
+    use std::fs::{self, File};
+    use std::io::{Read, Write};
+    use std::path::{Path, PathBuf};
+    use std::str;
+    use std::sync::atomic::*;
+
+    use git2;
```
Comment (ashleygwilliams) [inline tests/integration/main.rs]: this is unused.

File: src/main.rs [hunk 3]
```diff
@@ -38,7 +60,10 @@ main!(|args: Cli| {
...
     );
 
-    let _template = GitRepository::clone(&args.git, &project_dir)
+    let _template = RepoBuilder::new()
+        .bare(false)
+        .with_checkout(CheckoutBuilder::new())
+        .clone(&args.git, &project_dir)
         .with_context(|_e| format!("Couldn't clone `{}`", &args.git))?;
 
     fs::remove_dir_all(&project_dir.join(".git")).context("Error cleaning up cloned template")?;
```
Comment (steveklabnik) [inline src/main.rs]: The error here is a common issue with Windows and Git. The TL;DR is that git creates read-only files, which windows then refuses to remove; in essence, on NIX, you can delete a read-only file, but not on Windows.

An easy fix is to use https://crates.io/crates/remove_dir_all

Some sample code: https://github.com/steveklabnik/doxidize/blob/master/src/ops/clean.rs

Comment (steveklabnik) [reply src/main.rs]: I can confirm this fails locally. Trying out that crate:

```diff
diff --git a/src/main.rs b/src/main.rs
index cc8694a..61e87f8 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -7,6 +7,7 @@ extern crate indicatif;
 extern crate liquid;
 extern crate regex;
 extern crate walkdir;
+extern crate remove_dir_all;

 use dialoguer::Input;
 use git2::{
@@ -66,7 +67,7 @@ main!(|args: Cli| {
         .clone(&args.git, &project_dir)
         .with_context(|_e| format!("Couldn't clone `{}`", &args.git))?;

-    fs::remove_dir_all(&project_dir.join(".git")).context("Error cleaning up cloned template")?;
+    remove_dir_all::remove_dir_all(&project_dir.join(".git")).context("Error cleaning up cloned template")?;

```

(with the appropriate add in cargo.toml) errors for a different reason:

```
failures:

---- basics::simple_cargo_thing stdout ----
thread 'basics::simple_cargo_thing' panicked at 'Unexpected failure.
code-1
stderr=``````
command=`"git" "commit" "--message" "initial commit"`
code=1
stdout=```On branch master
nothing to commit, working tree clean
```
stderr=``````
', C:\Users\steve\.cargo\registry\src\github.com-1ecc6299db9ec823\assert_cmd-0.4.0\src\assert.rs:140:13
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by ashleygwilliams** [comment]
clsoing in favor of https://github.com/ashleygwilliams/cargo-generate/pull/20 thanks so much @killercup and @steveklabnik

**Pull request closed without merging**